scripting/include/srccoop/checkpoint.inc

```diff
@@ -149,6 +149,32 @@ enum struct CCoopEquipment
 	}
 }
 
+enum struct EquipmentManager 
+{
+    StringMap m_smEquipmentMap;
+
+
+    void Initialize()
+    {
+        this.m_smEquipmentMap = new StringMap();
+    }
+
+    void Clear()
```

Comment:
This one I'm not entirely sure what to do with. If I clear the handles in the store function I get invalid handle errors when I try to get the array later. I've implemented it for the clear function though. 

Which also raises another issue, I was trying to implement clear so that it would release the memory when a client disconnects via OnClientDisconnect but this was running in between map changes as well which broke the whole thing. Any ideas for how to handle client disconnection without deleting the equipment on map change?

Reply:
This should be resolved now. I hooked into player_disconnect event to dealloc the weapon list when a player disconnects. this event doesn't fire on map change and should handle all other disconnect events (such as timeout) from my research